src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/world/GenericBlockHighlighter.java

```diff
@@ -56,34 +56,68 @@ public void onWorldRenderLast(RenderWorldLastEvent event) {
 	@SubscribeEvent
 	public void onTick(TickEvent.ClientTickEvent ev) {
 		if (ev.phase != TickEvent.Phase.END) return;
-		highlightedBlocks.removeIf(it -> !isValidHighlightSpot(it) || !canPlayerSeeBlock(it.getX(), it.getY(), it.getZ()));
+		highlightedBlocks.removeIf(it -> !isValidHighlightSpot(it) ||
+			!canPlayerSeeNearBlocks(it.getX(), it.getY(), it.getZ()));
 	}
 
 	protected boolean canPlayerSeeBlock(double xCoord, double yCoord, double zCoord) {
 		EntityPlayerSP p = Minecraft.getMinecraft().thePlayer;
 		if (p == null) return false;
-		World w = p.worldObj;
-		MovingObjectPosition hitResult = w.rayTraceBlocks(
-			new Vec3(p.posX, p.posY + p.eyeHeight, p.posZ),
-			new Vec3(xCoord, yCoord, zCoord),
-			false,
-			true,
-			true
-		);
-		BlockPos bp = new BlockPos(xCoord, yCoord, zCoord);
+		Vec3 playerPosition = new Vec3(p.posX, p.posY + p.eyeHeight, p.posZ);
+		MovingObjectPosition hitResult = rayTraceBlocks(p.worldObj, playerPosition, xCoord, yCoord, zCoord);
```

Comment:
in this case i would probably change this to raytrace to the block center (`+0.5` on each coord)